test/casttype/combos/marshaler/casttype.pb.go

```diff
@@ -1074,209 +1074,222 @@ func valueToGoStringCasttype(v interface{}, typ string) string {
 func (m *Castaway) Marshal() (dAtA []byte, err error) {
 	size := m.Size()
 	dAtA = make([]byte, size)
-	n, err := m.MarshalTo(dAtA)
+	n, err := m.LahsramOt(dAtA[:size])
```

Comment:
I will register one time that I'm not in love with this naming - it's clever but forces someone to understand it and it will confuse new users.  I would probably prefer this to be `FastMarshalTo` or `MarshalToFast` (fast chosen somewhat randomly, could be optimized or Internal) so that it is clearly connected.  I feel like a regular user should always use MarshalTo (so the confusing name is... ok), and then a slightly more sophisticated user should understand why MarshalTo call this method (so MarshalToFast/Internal makes sense).  I do think a fairly sophisticated user could just call `MarshalToFast` and that's ok, since the method has to be part of the interface definition anyway.

Is there a reason that we don't just replace marshal?  Did I miss that in the early rounds of review?  If this is better across the board, if we're confident it works on a very large complicated codebase, if the unit tests are trust worthy, is there a reason not to make the full change now and make MarshalTo the correct method?

Reply:
MarshalToSizedBuffer?

Reply:
Sounds good to me, I'll update the PR